cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -323,6 +324,49 @@ private static void writeGenerators(Network network, Map<Terminal, String> mapTe
         }
     }
 
+    private static void writeBatteries(Network network, String cimNamespace, boolean writeInitialP,
+                                        XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        // Multiple synchronous machines may be grouped in the same generating unit
+        // We have to write each generating unit only once
+        Set<String> generatingUnitsWritten = new HashSet<>();
+        for (Battery battery : network.getBatteries()) {
+            String generatingUnit = context.getNamingStrategy().getCgmesIdFromProperty(battery, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "GeneratingUnit");
+            String reactiveLimitsId = null;
+            double minQ = 0.0;
+            double maxQ = 0.0;
+            switch (battery.getReactiveLimits().getKind()) {
```

Comment:
Maybe extract the code that exports the reactive capability curve to a separate function and call it for batteries and generators, to avoid duplicated code

Reply:
Refactoring done